### PR TITLE
[4.1] Support API save requests with adapter

### PR DESCRIPTION
**Pull request by laoneo**
Bugfix Pull Request for pr #35788.

### Summary of Changes
POST and PATCH are not working for the media web service when an adapter is set.

### Testing Instructions
```
curl --location --request PATCH 'https://[your J4 website]/api/index.php/v1/files/local-images:/joomla_black.png' \
--header 'Content-Type: application/json' \
--header 'X-Joomla-Token: [your user token]' \
--data-raw '{
    "path": "local-images:/joomla_black.png",
    "content": "[your base64 content]"
}'
```
### Actual result BEFORE applying this Pull Request
Returns 404 not found and image is not created.

### Expected result AFTER applying this Pull Request
Returns 200 and image contains the new created.

### CODE REVIEW

**Comment by alikon** [comment]
now it works with and without  adapters
image

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 1e73289f367ea7511afecbf60ede8776af61997e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36116">issues.joomla.org/tracker/joomla-cms/36116</a>.</sub>

File: tests/Codeception/api/com_media/MediaCest.php [hunk 1]
```diff
@@ -296,15 +296,64 @@ public function testCreateFolder(ApiTester $I)
...
+		);
+
+		$I->seeResponseCodeIs(HttpCode::OK);
+		$I->seeResponseContainsJson(['data' => ['attributes' => ['type' => 'file', 'name' => 'test.jpg']]]);
+	}
+
+	/**
+	 * Test the POST media files endpoint of com_media from the API with adapter information.
+	 *
+	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]: its probably not mixed is it... its `ApiTester` really... 

```suggestion
	 * @param   ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 2]
```diff
@@ -296,15 +296,64 @@ public function testCreateFolder(ApiTester $I)
...
+		);
+
+		$I->seeResponseCodeIs(HttpCode::OK);
+		$I->seeResponseContainsJson(['data' => ['attributes' => ['type' => 'dir', 'name' => 'test-from-create']]]);
+	}
+
+	/**
+	 * Test the PATCH media files endpoint of com_media from the API without adapter information.
+	 *
+	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param  ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 3]
```diff
@@ -350,15 +399,69 @@ public function testUpdateFolder(ApiTester $I)
 	}
 
 	/**
-	 * Test the DELETE media files endpoint of com_media from the API.
+	 * Test the PATCH media files endpoint of com_media from the API wit adapter information.
 	 *
 	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param  ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 4]
```diff
@@ -325,15 +374,15 @@ public function testUpdateFile(ApiTester $I)
 	}
 
 	/**
-	 * Test the PATCH media files endpoint of com_media from the API.
+	 * Test the PATCH media files endpoint of com_media from the API without adapter information.
 	 *
 	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param  ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 5]
```diff
@@ -350,15 +399,69 @@ public function testUpdateFolder(ApiTester $I)
...
+
+		$I->seeResponseCodeIs(HttpCode::OK);
+		$I->seeResponseContainsJson(['data' => ['attributes' => ['type' => 'file', 'name' => 'override.jpg']]]);
+		$I->dontSeeResponseContainsJson(['data' => ['attributes' => ['content' => '1']]]);
+	}
+
+	/**
+	 * Test the PATCH media files endpoint of com_media from the API with adapter information.
+	 *
+	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param  ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 6]
```diff
@@ -350,15 +399,69 @@ public function testUpdateFolder(ApiTester $I)
...
+		);
+
+		$I->seeResponseCodeIs(HttpCode::OK);
+		$I->seeResponseContainsJson(['data' => ['attributes' => ['type' => 'dir', 'name' => 'override-new']]]);
+	}
+
+	/**
+	 * Test the DELETE media files endpoint of com_media from the API without adapter information.
+	 *
+	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param   ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 7]
```diff
@@ -389,6 +492,46 @@ public function testDeleteFolder(ApiTester $I)
 		$I->seeResponseCodeIs(HttpCode::NO_CONTENT);
 	}
 
+	/**
+	 * Test the DELETE media files endpoint of com_media from the API with adapter information.
+	 *
+	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param   ApiTester  $I  Api tester
```

File: tests/Codeception/api/com_media/MediaCest.php [hunk 8]
```diff
@@ -389,6 +492,46 @@ public function testDeleteFolder(ApiTester $I)
...
+		$I->haveHttpHeader('Accept', 'application/vnd.api+json');
+		$I->sendDelete('/media/files/local-images:/' . $this->testDirectory . '/todelete.jpg');
+
+		$I->seeResponseCodeIs(HttpCode::NO_CONTENT);
+	}
+
+	/**
+	 * Test the DELETE media files endpoint of com_media from the API with adapter information.
+	 *
+	 * @param   mixed   ApiTester  $I  Api tester
```
Comment (PhilETaylor) [inline tests/Codeception/api/com_media/MediaCest.php]:
```suggestion
	 * @param   ApiTester  $I  Api tester
```

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 2630c174785be3b13d43d3087e643fc66ae93b6b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36116">issues.joomla.org/tracker/joomla-cms/36116</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36116">issues.joomla.org/tracker/joomla-cms/36116</a>.</sub>

**Comment by bembelimen** [comment]
Thanks

**Pull request merged**